src/Agent.Worker/Release/ReleaseFileSystemManager.cs

```diff
@@ -117,6 +130,13 @@ public string JoinPath(string rootDirectory, string relativePath)
             {
                 await stream.CopyToAsync(targetStream, StreamBufferSize, cancellationToken);
             }
+            
+            if (sourceFileInfo != null)
+            {
+                var targetFileInfo = GetFileInfo(filePath);
+                targetFileInfo.CreationTime = sourceFileInfo.CreationTime;
+                targetFileInfo.LastWriteTime = sourceFileInfo.LastWriteTime;
```

Comment:
don't see the time stamp updates to targetFileInfo getting applied back. are you sure it gets applied back?